File: GitToolBox/src/main/java/zielu/intellij/util/ZResBundle.java
package zielu.intellij.util;

import com.intellij.BundleBase;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.PropertyKey;

public final class ZResBundle {
  @NonNls
  private static final String BUNDLE_NAME = "zielu.intellij.ZResBundle";
  private static final ZBundleHolder BUNDLE_HOLDER = new ZBundleHolder(BUNDLE_NAME);

  private ZResBundle() {
    throw new IllegalStateException();
  }

  @NotNull
  public static String message(@NotNull @PropertyKey(resourceBundle = BUNDLE_NAME) String key, Object... params) {
    return BundleBase.message(BUNDLE_HOLDER.getBundle(), key, params);
  }
}

File: GitToolBox/src/main/kotlin/zielu/intellij/util/ZResBundle.kt
package zielu.intellij.util

import com.intellij.BundleBase
import org.jetbrains.annotations.NonNls
import org.jetbrains.annotations.PropertyKey

internal object ZResBundle {
  @NonNls
  private const val BUNDLE_NAME = "zielu.intellij.ZResBundle"
  private val BUNDLE_HOLDER = ZBundleHolder(BUNDLE_NAME)

  fun message(@PropertyKey(resourceBundle = BUNDLE_NAME) key: String, vararg params: Any?): String {
    return BundleBase.message(BUNDLE_HOLDER.bundle, key, *params)
  }
}
